pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -80,6 +93,10 @@ def distributed_sampler_kwargs(self):
         distributed_sampler_kwargs = dict(num_replicas=(self.num_nodes * self.num_processes), rank=self.global_rank)
         return distributed_sampler_kwargs
 
+    @property
+    def _is_single_process_single_device(self) -> bool:
```

Comment:
No need for private property there.

Reply:
And let's add this one to all training_type_plugin.

Reply:
then probably could add this to `ParallelPlugin`

Reply:
@tchaton SPMD DDP will be deprecated from PyTorch so I don't think we should expose this property to more training type plugins in lightning: https://github.com/pytorch/pytorch/issues/47012